cheddar/cheddar-events/src/main/java/com/clicktravel/cheddar/event/EventMessageHandler.java

```diff
@@ -40,20 +42,30 @@ public void handle(final TypedMessage typedMessage) throws MessageHandlingExcept
         try {
             final String messageType = typedMessage.getType();
             logger.debug("Handling: " + messageType);
-            final EventHandler<E> eventHandler = eventHandlers.get(messageType);
-            final E event = AbstractEvent.newEvent(eventHandler.getEventClass(), typedMessage.getPayload());
-            if (!eventHandler.getEventClass().isAssignableFrom(event.getClass())) {
-                throw new IllegalStateException("Event of type " + event.getClass() + " is not compatible with "
-                        + eventHandler.getEventClass() + " in event handler");
+            for (final EventHandler<E> eventHandler : eventHandlers.get(messageType)) {
+                try {
+                    final E event = AbstractEvent.newEvent(eventHandler.getEventClass(), typedMessage.getPayload());
+                    if (!eventHandler.getEventClass().isAssignableFrom(event.getClass())) {
+                        throw new IllegalStateException("Event of type " + event.getClass()
+                                + " is not compatible with " + eventHandler.getEventClass() + " in event handler");
+                    }
+                    eventHandler.handle(event);
+                } catch (final Exception e) {
+                    logger.error("Error handling message: " + typedMessage, e);
```

Comment:
- The ```MessageHandlingException``` catch block below can still be hit as ```TypedMessage#getType``` can throw a ```MessageParseException``` (see class ```InvalidTypedMessage```)
- The retry logic is not affected. ```@Retryable``` methods called during event handling will perform any needed retries before giving up and throwing an exception that will be caught here. Prior to this change, the exception would have been caught in ```MessageHandlerWorker```, which simply logs a ERROR message. That behaviour has been copied over to the new catch block here.